jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ExtraDirectoriesParameters.java

```diff
@@ -92,10 +96,28 @@ public List<ExtraDirectoryParameters> getPaths() {
    * @param paths paths to set.
    */
   public void setPaths(Object paths) {
-    this.paths.set(
-        project.files(paths).getFiles().stream()
-            .map(file -> new ExtraDirectoryParameters(objects, project, file.toPath(), "/"))
-            .collect(Collectors.toList()));
+    this.paths.set(transformObjectToParametersList(paths));
+  }
+
+  /**
+   * Sets paths, for lazy evaluation where {@code paths} is a {@link Provider} of a suitable object.
+   *
+   * @param paths provider of paths to set
+   * @see #setPaths(Object)
+   */
+  public void setPaths(Provider<Object> paths) {
+    this.paths.set(paths.map(this::transformObjectToParametersList));
+  }
+
+  /**
+   * Helper method to convert {@code Object} to {@code List<ExtraDirectoryParameters>} in {@code
+   * setFrom}.
+   */
+  @Nonnull
+  private List<ExtraDirectoryParameters> transformObjectToParametersList(Object obj) {
```

Comment:
Since the parameter of this method already gives us an indication of what we are converting, how about we do the following?

```suggestion
  private List<ExtraDirectoryParameters> convertToExtraDirectoryParametersList(Object obj) {
```

Reply:
Sounds good!